pkg/prometheus_ingester/prometheus_ingester_test.go

```diff
@@ -718,3 +720,65 @@ func Test_processHistogramIncrease(t *testing.T) {
 		assert.ElementsMatchf(t, testCase.expectedEvents, generatedEvents, "expected events:\n%s\n\nresult:\n%s", testCase.expectedEvents, generatedEvents)
 	}
 }
+
+func TestPrometheusIngesterHttpHeader_UnmarshalYAML(t *testing.T) {
+	const envName = "ENVNAME"
+	const envValue = "envValue"
+
+	if err := os.Setenv(envName, envValue); err != nil {
+		t.Fatal(err)
+	}
+
+	tests := []struct {
+		name          string
+		goldenfile    string
+		expectedValue string
+		wantErr       bool
+	}{
+		{
+			name:          "value from string",
+			goldenfile:    "goldenfiles/prometheus_ingester_httpheader_unmarshal1.yaml",
+			expectedValue: "stringValue",
+			wantErr:       false,
```

Comment:
Nit: Maybe keep just either expectedValue or wantErr -- to keep this brief. Repeating default values does not add any value here, I think.

Reply:
fixed d48902d4ad77ea48e830ce5ce24c2a6aa5fce662